mdnssd-min.c

```diff
@@ -1015,11 +1017,33 @@ static mDNSservice_t *build_update(struct context_s *context, bool build) {
 		s->status = MDNS_CURRENT;
 	}
 
-	// now we can remove the service
-	if (expired) {
+	if (ptr_expired && srv_expired && txt_expired) {
```

Comment:
I have to think about that, it was a long time but I vaguely remember that requiring all services to be expired was causing issues like *very* long delay for some services to disappear as some RR have high timeouts. So the detection of missing services was not detected

Reply:
The change is to remember data about the still-valid RRs in context->slist, rather than discarding then whenever one RR expires. 

Note, however, the the expiry event is emitted whenever any of the three RRs expires. (This happens around like 990.)

Reply:
right ... sorry I don't even remember my own code from a while ago ...